### PR TITLE
Change sourcelocale to string, with culture info validation

### PR DESCRIPTION
**Pull request by wangrui-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs [hunk 1]
```diff
@@ -34,15 +34,25 @@ public TextSource(string text, string voiceName)
...
+        /// <param name="sourceLocale"> The culture info string of the voice. </param>
         /// <param name="gender"> The gender of the voice. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="text"/> is null. </exception>
-        public TextSource(string text, CultureInfo sourceLocale, GenderType gender)
+        /// <exception cref="ArgumentException"> <paramref name="sourceLocale"/> is wrong format. </exception>
+        public TextSource(string text, string sourceLocale, GenderType gender)
         {
             Argument.AssertNotNull(text, nameof(text));
 
+            try
```
Comment (Paresh-Arvind-Patil) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs:45]: @wangrui-msft  Should we add the same check for SpeechLanguage?

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs:45]: The speech language doesn't have the constructor, do you suggest to add a new constructor for speech language?

File: sdk/communication/Azure.Communication.CallAutomation/tests/CallMedias/CallMediaTests.cs [hunk 2]
```diff
@@ -319,5 +328,9 @@ public void MediaOperations_Return404NotFound(Func<CallMedia, Response<StartReco
                 }
             };
         }
+        private static void TestWrongTestSource_PlayOperations()
+        {
+            TextSource _wrongTextSource = new TextSource("PlayTTS test text.", "en-US-ElizabethNeural", GenderType.Female);
```
Comment (Paresh-Arvind-Patil) [inline sdk/communication/Azure.Communication.CallAutomation/tests/CallMedias/CallMediaTests.cs:333]: Can you add invalid source locale such as hi-US (hindi-US) or ar-US (arabic-US) and check if this is caught by culture-info? 

Just for your information: Here is the list of valid source locale supported by cognitive services : https://learn.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support?tabs=stt

Comment (Paresh-Arvind-Patil) [reply sdk/communication/Azure.Communication.CallAutomation/tests/CallMedias/CallMediaTests.cs:333]: I just want to check if this is something CultureInfo can validate?  This is not a PR blocker.

Comment (wangrui-msft) [reply sdk/communication/Azure.Communication.CallAutomation/tests/CallMedias/CallMediaTests.cs:333]: ok, will try to add some more cases

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs [hunk 3]
```diff
@@ -34,15 +34,15 @@ public TextSource(string text, string voiceName)
 
         /// <summary> Initializes a new instance of TextSource. </summary>
         /// <param name="text"> Text for the cognitive service to be played. </param>
-        /// <param name="sourceLocale"> The culture info of the voice. </param>
+        /// <param name="sourceLocale"> The culture info string of the voice. </param>
```
Comment (Paresh-Arvind-Patil) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/TextSource.cs:37]: Remove "culture info" text ?

**Review by Paresh-Arvind-Patil — APPROVED** [review]

**Pull request merged**